Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs

```diff
@@ -57,32 +66,45 @@ public override Vector3 ApplyFriction(Vector3 velocity)
     {
         base.ApplyInputs(inVelocity, inExternalForces, out outVelocity, out outExternalForces);
 
+        //normalize the direction so that it just gives the wall jump angle
+        direction = player.AimDirection.normalized;
         int wallDirX = (controller.collisions.Left) ? -1 : 1;
-        float moveX = player.InputPlayer.GetAxis("Move Horizontal");
 
         jumped = false;
+
+        Debug.Log("direction x: " + direction.x + " direction y: " + direction.y);
         if (player.InputPlayer.GetButtonDown("Jump"))
         {
             jumped = true;
             jumpSound.Play();
 
-            //if the input is pointed towards the wall
-            if (Mathf.Abs(wallDirX - moveX) < 0.5f)
+            //PLayer will always jump off of a wall a little bit
+            outVelocity.x = (wallJumpVelocity.x / 2) * (-wallDirX) ;
+
+            //if player has no input
+            if (direction.x == 0)
+            {
+                outVelocity.x = -wallDirX * wallJumpVelocity.x * 2;
+                outVelocity.y = wallJumpVelocity.y;
+
+            }
+            //if the input is pointed towards the wall or slightly away and up
+            else if (Mathf.Abs(wallDirX - direction.x) < 0.5f ||  ((Mathf.Abs(wallDirX - direction.x) < 1.16) && (Mathf.Abs(1 - direction.y) < .25 )))
             {
-                outVelocity.x = -wallDirX * wallJumpClimbForces.x;
-                outVelocity.y = wallJumpClimbForces.y;
+                outVelocity.x = (-wallDirX) * (wallJumpVelocity.x / 2);
+                outVelocity.y = wallJumpVelocity.y;
             }
-            //if the player has no x input
-            else if (moveX == 0)
+            //if the input is down and slightly away
+            else if(Mathf.Abs(wallDirX - direction.x) < 1.16  && Mathf.Abs(1 - direction.y) > .25)
```

Comment:
I really don't like having magic numbers like this. These should be moved out into variables that describe what they actually are.